ncm-nss/src/main/perl/nss.pm

```diff
@@ -95,28 +95,28 @@ sub Configure {
 
     my @new = ( "# Generated by ncm-nss\n\n" );
     push(@new, map { "$_: $databases->{$_}\n" } sort keys %$databases);
-    my $contents = join("", @new);
-    # cannot use CAF::FileWriter because close() always returns false in
-    # noaction mode so cannot show when nscd would be restarted
-    my $result = LC::Check::file("/etc/nsswitch.conf",
-                                 backup => ".OLD",
-                                 contents => $contents,
-                                 owner => "root",
-                                 group => "root",
-                                 mode => "0444",
-                                );
+    my $fh = CAF::FileWriter->open("/etc/nsswitch.conf",
+                                   backup => ".OLD",
+                                   owner => "root",
+                                   group => "root",
+                                   mode => 0444,
+                                   log => $self,
+        );
+    print $fh join("", @new);
+    my $result = $fh->close();
+
     if ($result) {
-        $self->log("updated /etc/nsswitch.conf");
+        $self->info("updated /etc/nsswitch.conf") unless $NoAction;
```

Comment:
this looks like some idiom to add to CAF, but also related to https://github.com/quattor/CAF/issues/100.
in any case, most components do not make this distinction (and it should be handeled on e.g. CAF level).

anyway, what is interesting is that the tests pass without any definition of `$NoAction` (i expected `use strict` or `use warning` to at least make some noise)